libs/hdf-converters/src/aws-config-mapper.ts

```diff
@@ -0,0 +1,334 @@
+import {
+  ComplianceByConfigRule,
+  ConfigRule,
+  ConfigService,
+  ConfigServiceClientConfig,
+  DescribeConfigRulesCommandInput,
+  DescribeConfigRulesCommandOutput,
+  EvaluationResult
+} from '@aws-sdk/client-config-service';
+import {ExecJSON} from 'inspecjs';
+import _ from 'lodash';
+import path from 'path';
+import {version as HeimdallToolsVersion} from '../package.json';
+import {AwsConfigMapping} from './mappings/AwsConfigMapping';
+
+const NOT_APPLICABLE_MSG =
+  'No AWS resources found to evaluate compliance for this rule';
+const INSUFFICIENT_DATA_MSG =
+  'Not enough data has been collected to determine compliance yet.';
+const NAME = 'AWS Config';
+
+const AWS_CONFIG_MAPPING_FILE = path.resolve(
+  __dirname,
+  '../data/aws-config-mapping.csv'
+);
+const AWS_CONFIG_MAPPING = new AwsConfigMapping(AWS_CONFIG_MAPPING_FILE);
+
+export class AwsConfigMapper {
+  configService: ConfigService;
+  issues: Promise<ConfigRule[]>;
+  results: ExecJSON.ControlResult[][];
+  constructor(options: ConfigServiceClientConfig) {
+    this.configService = new ConfigService(options);
+    this.results = [];
+    this.issues = this.getAllConfigRules();
+  }
+  private async getAllConfigRules(): Promise<ConfigRule[]> {
+    let params: DescribeConfigRulesCommandInput = {
+      ConfigRuleNames: [],
+      NextToken: ''
+    };
+    const configRules: ConfigRule[] = [];
+    let response = await this.getConfigRulePage(params);
+    if (response.ConfigRules === undefined) {
+      throw new Error('No data was returned');
+    } else {
+      while (response !== undefined && response.ConfigRules !== undefined) {
+        response.ConfigRules.forEach((rule) => {
+          configRules.push(rule);
+        });
+        if (response.NextToken) {
+          params = _.set(params, 'NextToken', response.NextToken);
+        } else {
+          break;
+        }
+        response = await this.getConfigRulePage(params);
+      }
+    }
+    this.results = await this.getResults(configRules);
+    return configRules;
+  }
+  private async getConfigRulePage(
+    params: DescribeConfigRulesCommandInput
+  ): Promise<DescribeConfigRulesCommandOutput> {
+    return this.configService.describeConfigRules(params);
+  }
+  private async getResults(
+    configRules: ConfigRule[]
+  ): Promise<ExecJSON.ControlResult[][]> {
+    const complianceResults: ComplianceByConfigRule[] =
+      await this.fetchAllComplianceInfo(configRules);
+    const results = configRules.map(async (rule) => {
```

Comment:
This map call will simultaneously make requests for every rule, which in any normal use case will almost always result in a throttling exception from AWS. This is a limitation that AWS places on users, so this should be changed to a simple for loop where rules are queried sequentially.